paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -144,7 +147,13 @@ public FlinkCatalog(
             ClassLoader classLoader,
             Options options) {
         super(name, defaultDatabase);
-        this.catalog = catalog;
+        boolean cacheEnabled = options.get(CACHE_CATALOG);
+        long cacheExpirationIntervalMs = options.get(CACHE_CATALOG_EXPIRATION_MS);
+        if (cacheEnabled) {
```

Comment:
Shall we add this in CatalogFactory

Reply:
If the catalog is only created once, this code can be moved to `CatalogFactory`. However, in some test case of the `flink-cdc` module, they use the same `warehouse` to create different catalogs. If a cache catalog is added, it will result in assert failure.